app/models/user.rb

```diff
@@ -1,10 +1,20 @@
 class User < ApplicationRecord
-  before_save{self.email = email.downcase}
-  validates :name, presence: true, length: {maximum: 50}
+  before_save{email.downcase}
```

Comment:
`{email.downcase}` chú đã test chứ, nghi ngờ hơm hề nhẹ nhé
đúng phải là `{email.downcase!}`

phân biệt cho chị `downcase!` và `downcase` nhé

Reply:
downcase! tức là khi chuỗi của chị nó đã là chũ thườn thì nó trả về nil vì ko cso j thay đổi cả! còn khi chuỗi của chị có kí tự hoa thì nó sẽ trả về chuỗi kí tự thường
downcase thì nó xử lý chuỗi như bình thường! nếu chuỗi có kí tự hoa hay ko hoa j nữa thì nó vẫn sẽ trả về 1 chuỗi kí tự thường!

Reply:
Câu trả lời chưa đầy đủ, trả lời lại sự khác nhau trong các trường hợp sau:
- Kết quả trả về trong trường hợp không có kí tự in hoa?
- Sự thay đổi của chuỗi gốc dùng gọi method?

Reply:
1 kết quả trả về khi ko có kí tự in hoa
        downcase! trả về nil   
        downcase trả về chuỗi gốc của nó
2 sự thay đổi của chuỗi gốc khi dùng method
        em chưa hiểu câu hỏi lắm!